api/maven-api-core/src/main/java/org/apache/maven/api/services/ArtifactCoordinateFactory.java

```diff
@@ -43,6 +43,18 @@ public interface ArtifactCoordinateFactory extends Service {
     @Nonnull
     ArtifactCoordinate create(@Nonnull ArtifactCoordinateFactoryRequest request);
 
+    /**
+     * Creates a coordinate out of string that is formatted like:
+     * {@code <groupId>:<artifactId>[:<extension>[:<classifier>]]:<version>}
+     *
+     * @param session the session.
+     * @param coordinateString the string having "standard" coordinate.
+     * @return an {@code Artifact}, never {@code null}
+     * @throws IllegalArgumentException if {@code request} is null or {@code request.session} is null or invalid
+     */
+    @Nonnull
+    ArtifactCoordinate create(@Nonnull Session session, @Nonnull String coordinateString);
```

Comment:
In other services, when multiple "signatures" are involved, a single "request" object is used to store all those bits.  I'm not sure why we should not do the same here, i.e. add a `coordinateString` member to the `ArtifactCoordinateRequest`.

Reply:
moved stuff into request